src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -733,16 +734,17 @@ public static bool TryCompleteSendTo(SafeCloseSocket socket, ReadOnlySpan<byte>
 
                 if (sent == -1)
                 {
-                    if (errno != Interop.Error.EAGAIN && errno != Interop.Error.EWOULDBLOCK)
+                    if (!successfulSend && errno != Interop.Error.EAGAIN && errno != Interop.Error.EWOULDBLOCK)
```

Comment:
If I understand this, it looks like we're going to eat any error if successfulSend = true.  I don't think that's what we actually want to do, is it?  We just want to return Success instead of WouldBlock if we were able to send any data.  Right?

Reply:
I would think we would want to.  If we successfully sent 10 bytes but then got some kind of error trying to send the next 5, shouldn't we be communicating that we successfully sent the 10?

Reply:
We shouldn't swallow 'any' success. Failures find their way back to you :smile: